base/src/main/java/org/aya/core/term/Term.java

```diff
@@ -84,64 +174,34 @@ default VarConsumer.ScopeChecker scopeCheck(@NotNull ImmutableSeq<LocalVar> allo
    *              Can be null only if we're absolutely sure that holes are frozen,
    *              like in the error messages.
    */
-  default @NotNull Term normalize(@NotNull TyckState state, @NotNull NormalizeMode mode) {
-    if (mode == NormalizeMode.NULL) return this;
-    if (mode == NormalizeMode.NF) return this.view().normalize(state).commit();
-    return accept(new Normalizer(state), mode);
+  default @NotNull Term normalize(@Nullable TyckState state, @NotNull NormalizeMode mode) {
```

Comment:
Why nullable state?

Reply:
The `normalize` method are used in places where `state` can be nullable, and the actual implementation do handle the case for null states.
I think I mislabeled it to be `NotNull` previously.